torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -282,23 +269,9 @@ void RequestCallbackImpl::processScriptRemoteCall(
     }
   };
 
-  c10::intrusive_ptr<c10::ivalue::Future> jitFuture;
-  try {
-    jitFuture = PythonRpcHandler::getInstance()
-                    .jitCompilationUnit()
-                    ->get_function(scriptRemoteCall.qualifiedName())
-                    .runAsync(stack);
-    if (jitFuture->completed()) { // short-cut.
```

Comment:
we are skipping fast passes, let's measure the impact.

Reply:
What benchmarks would you like me to run?

Also, I don't remember if I said it clearly before, so let me explain why I think it's better to avoid these fastpaths/shortcuts: they are terribly dangerous for CUDA sync, because at no point we're synchronizing the events contained in the future with the current streams at the callsite. (It can be done, one just has to call `future->wait()`, but we didn't do it anywhere, and it's super easy to forget about).

Moreover, I believe they hurt readability, because they force us to lay out the code in a different order than how it's executed. Namely, now we need to do something like this:
```
continuation = []() { bar(); };
auto fut = foo();
if (fut->completed()) {
  continuation();
} else {
  fut->addCallback(continuation);
}
```
Note how in the code `bar()` appears _before_ `foo()`, whereas in the "logical flow" of the program `bar()` comes _after_ `foo()`. This requires the reader to "jump around" and makes it harder to quickly understand what the code is doing. Without shortcuts, the above code becomes:
```
auto fut = foo();
fut->addCallback([]() { bar(); });
```

As for the supposed performance benefit of these fastpaths, I honestly don't think that we're introducing a regression. The `addCallback` method itself already has such a shortcut logic internally, so in fact these fastpaths were redundant. Moreover, after the changes in #57638, invoking `addCallback` should come with literally zero overhead, and the compiler will most likely inline it all the time. (I said `addCallback` but I also mean `then` since they're basically the same).

Reply:
IIRC, @jjlilley added this a while ago when optimizing PyPer perf. So, I assume there was a reason to introduce fast path at least at that time. 

> Moreover, after the changes in #57638, invoking addCallback should come with literally zero overhead, and the compiler will most likely inline it all the time. 

If we are confident there is no additional overhead introduce, this looks good to me.